src/replication.c

```diff
@@ -216,20 +204,21 @@ void rebaseReplicationBuffer(long long base_repl_offset) {
     }
 }
 
-/* Replication: Master side - connection peering.
- * On rdb-channel sync, connection peering is used to keep replication data in 
- * the backlog until the replica requests PSYNC. Peering happens in two forms, 
+/* Replication: Master side - connections association.
+ * On rdb-channel sync, connection association is used to keep replication data in 
+ * the backlog until the replica requests PSYNC. Association happens in two forms, 
  * if there's an existing buffer block at the fork time, the replica is attached
  * to the tail, if there is no tail, the replica will be attached when a new
  * buffer block is created (see the Retrospect function below).
-  * Replica listening IP and port is used as a unique key
- * for the peering. On COB overrun, peering is deleted and the RDB connection 
+ * Replica rdb client id is used as a unique key for the association. 
+ * On COB overrun, association is deleted and the RDB connection 
  * is dropped.
  */
-void peerPendingSlaveToBacklogBlock(client* slave) {
+void addSlaveToPsyncWaitingDict(client* slave) {
     listNode *ln = NULL;
     replBufBlock *tail = NULL;
-    sds slave_name = sdsnew(replicationGetSlaveNameGeneric(slave, 0));
+    uint64_t *cid = (uint64_t*)zmalloc(sizeof(uint64_t));
```

Comment:
AFAIK the key to the dict can be the integer itself, no need for heap allocation for it.

Reply:
True, I just think it a bit odd to fool the dictionary that cid is a pointer while it is actually the key itself. It might cause confusion, as well as threaten memory access.

Reply:
i don't see the problem. the dictType can have keyCompare, keyDup, and keyDestructor set to NULL.
and the keys are always plain integers (casted to void*).
the hashFunction can either just return the integer, or use the more sophisticated dictIntHashFunction mentioned above (not sure about it's benefits)

Reply:
Ack, sounds good. So dict key will be plain integers and if we don't have a reason to use mask then the hash will be the identity function.